.github/workflows/deploy-dev-pkgs.yml

```diff
@@ -0,0 +1,82 @@
+name: Deploy "conda" conda packages
```

Comment:
Since the name is used in the actions UI could we just call this "Canary" and rename the yml file to canary.yml? Just a nit but "deploy" has a particular meaning on GitHub and we're not doing that here.

Reply:
Good point; how about just "Build & upload"? ("Publish", it turns out, also has a specific meaning in GitHub.)

I purposely avoided the word "Canary" since forks of this repo won't have permissions to upload packages to that channel, but this GH Action should still be usable by forks (as long as they have their `ANACONDA_*` repository secrets properly configured).